src/gas/gas-util.ts

```diff
@@ -0,0 +1,104 @@
+import { BN } from 'ethereumjs-util';
+import { query, handleFetch, gweiDecToWEIBN, weiHexToGweiDec } from '../util';
+import {
+  GasFeeEstimates,
+  EthGasPriceEstimate,
+  EstimatedGasFeeTimeBounds,
+  unknownString,
+  LegacyGasPriceEstimate,
+} from './GasFeeController';
+
+const GAS_FEE_API = 'https://mock-gas-server.herokuapp.com/';
+export const EXTERNAL_GAS_PRICES_API_URL = `https://api.metaswap.codefi.network/gasPrices`;
+
+export async function fetchGasEstimates(): Promise<GasFeeEstimates> {
+  return await handleFetch(GAS_FEE_API);
+}
+
+/**
+ * Hit the legacy MetaSwaps gasPrices estimate api and return the low, medium
+ * high values from that API.
+ */
+export async function fetchLegacyGasPriceEstimates(): Promise<LegacyGasPriceEstimate> {
+  const result = await handleFetch(EXTERNAL_GAS_PRICES_API_URL, {
+    referrer: EXTERNAL_GAS_PRICES_API_URL,
+    referrerPolicy: 'no-referrer-when-downgrade',
+    method: 'GET',
+    mode: 'cors',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  return {
+    low: result.SafeGasPrice,
+    medium: result.ProposeGasPrice,
+    high: result.FastGasPrice,
+  };
+}
+
+export async function fetchEthGasPriceEstimate(
+  ethQuery: any,
+): Promise<EthGasPriceEstimate> {
+  const gasPrice = await query(ethQuery, 'gasPrice');
+  return {
+    gasPrice: weiHexToGweiDec(gasPrice).toString(),
```

Comment:
This seems to be failing: `Invalid character in 0x5f48b0f7`

Reply:
Maybe without the `0x`?

Reply:
aabfa3a good catch, added a stripHexPrefix in the conversion method and added a test